# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../libdomainkeys/libdomainkeys-segv.patch
# Copyright (C) 2009 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

diff -urN libdomainkeys-0.66.org/domainkeys.c libdomainkeys-0.66/domainkeys.c
--- libdomainkeys-0.66.org/domainkeys.c	2005-06-21 04:55:39.000000000 +0200
+++ libdomainkeys-0.66/domainkeys.c	2005-06-25 17:04:10.498813408 +0200
@@ -1384,11 +1384,14 @@
   }
   if (s && *s && s[1] && s[1] != '@')
   {
-    s = strchr(s, '@') + 1;
-    if (*s == '\0')
-    {
-      //make sure we have an actual domain after @
-      s = NULL;
+    s = strchr(s, '@');
+    if (s) {
+	s += 1;
+	if (*s == '\0')
+	{
+	    //make sure we have an actual domain after @
+	    s = NULL;
+	}
     }
   }
   else
